### PR TITLE
adminguide: add overview, update for 0.31.0, improve flow

### PR DESCRIPTION
**Pull request by garlick**
I'll call this a WIP for now because I doubt it's ready to merge, but any feedback would be greatly appreciated.

I'm starting to go cross-eyed so thought I'd better park this and do something else for a while.

### CODE REVIEW

**Comment by garlick** [comment]
I'm going to step away from this for a while so if anybody wants to have a peek and provide feedback, that would be helpful.

I added a really lame diagram featuring Flux frog and fox.  Did I say it was really lame?  I'm ready for your insults, or better yet, suggestions for improvement.

**Comment by garlick** [comment]
This is ready for a review.  Probably the best strategy is to just read the new doc focusing on the new overview section, and the flux-accounting section which was pared down quite a bit from what was merged last week.

File: adminguide.rst [hunk 1]
```diff
@@ -312,136 +356,104 @@ This space should be preserved across a reboot as it contains the Flux
...
-.. code-block:: console
+- The job-archive module scans inactive jobs and dumps them to a sqlite
+  database.
+- A script reads the archive database and updates the job usage data in the
+  accounting database.
+- A script updates the per-user fair share factors in the accounting database.
+- A script pushes updated factors to the multi-factor priority plugin.
 
- $ flux account update-usage
+The ``job-archive`` module must be configured to run periodically:
```
Comment (cmoussa1) [inline adminguide.rst:436]: I like the inclusion of the `job-archive` module here, thanks for including this. I think this drives home the use for the `update-usage` command and how it works. Perhaps in the near future, flux-accounting can have more detailed descriptions on a separate page of how it calculates job priorities using the archive DB?

**Review by cmoussa1 — COMMENTED** [review]
Thanks for taking the time to revise this documentation and improve its flow. I read the flux-accounting section a couple of times and I think its improvements are huge. It reads much more concise and to the point, and I think it leaves room for me to expand on some of the details of how it calculates job priority, job usage, fairshare, etc. on a separate page or something in the near future. Great work so far!

I haven't taken a look at the rendered page just yet, but will do that shortly.

File: adminguide.rst [hunk 2]
```diff
@@ -125,97 +182,92 @@ In this guide, separate files will typically be used for clarity, instead
...
-
 By default, a Flux instance does not allow access to any user other than
 the instance *owner*, in this case the ``flux`` user.  This is not
-suitable for a system instance, so *guest user* access should be enabled
-in ``/etc/flux/system/conf.d/access.toml``.  In addition, it may be convenient
-to allow the ``root`` user to act as the instance owner, to give system
-administrators privileged Flux access to cancel or list jobs:
+suitable for a system instance, so *guest user* access should be enabled.
+In addition, it may be convenient to allow the ``root`` user to act as the
+instance owner, to give system administrators privileged Flux access to
```
[the remark this answers is not in the document]
Comment (garlick) [reply adminguide.rst:227]: Honestly it works for me both ways but I'm happy to drop the comma :-)

[the remark this answers is not in the document]
Comment (garlick) [reply adminguide.rst]: That sentence was not great.  I replaced it - hopefully it flows better now:

> In addition, for convenience, the ``root`` user should be allowed to act n the role of instance owner:

[the remark this answers is not in the document]
Comment (cmoussa1) [reply adminguide.rst]: I like the new sentence. Thanks!

**Review by cmoussa1 — APPROVED** [review]
Thanks @garlick! I had a chance to review the rendered document and I think it looks great. LGTM.

**Comment by garlick** [comment]
Thanks!  I know everyone is busy so I'll plan on setting MWP after lunch, unless someone says they want to review before then.

**Comment by grondo** [comment]
I had reviewed before the second round of edits and was planning another look. However, my conclusion with the first cut was that it was much improved. I'll take a quick look but don't anticipate any comments, and we can always add more edits in future PRs.

What I think we need next is a reference guide for the configuration files from various projects. The guides should probably be sited within the project they describe, so that new configuration options can be documented along with the PR that adds them, but they should all be linked into a "Reference" section in the overall RTD documentation.

Perhaps the right approach here is to document config files in a single, per-project, manpage. Or maybe just an RST doc meant for the web would be simpler, I have no idea. Then, this admin guide can document the bare essentials for setting up a system instance and reference the appropriate config file documentation section for further detail.

**Comment by garlick** [comment]
Excellent idea.  It seems like man page links are appropriate, at a minimum.  There is a reference to [flux-config-bootstrap(5)](https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man5/flux-config-bootstrap.html) already.  Should we make that the convention, that projects populate section 5 with config file info that we can link to from this document?  I'd be fine with renaming that page to `flux-core-config(5)` or similar if we wanted to go with one page project.

Setting MWP per discussion offline.

**Pull request merged**